lang/cs/Org.Apache.REEF.Network/Elastic/Operators/Physical/IElasticIterator.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Utilities.Attributes;
+using System;
+
+namespace Org.Apache.REEF.Network.Elastic.Operators.Physical
+{
+    /// <summary>
+    /// Group communication operator used to for iterations.
+    /// </summary>
+    [Unstable("0.16", "API may change")]
+    public interface IElasticIterator : IElasticOperator
+    {
+        /// <summary>
+        /// Move to the next iteration.
+        /// </summary>
+        /// <returns>True if the next iteration exists</returns>
+        bool MoveNext();
+
+        /// <summary>
+        /// The current iteration.
+        /// </summary>
+        /// <returns>An object representing the current iteration</returns>
+        object Current { get; }
```

Comment:
why not just implement `IEnumerator` that gives us `MoveNext()`, `Current`, and `Reset()`, and plays well with other C# features like `foreach` operator?

Reply:
Can we postpone this topic to when I will push the iterator implementation? The intuition is that there can be 2 type of iterators: _enumerable_ iterators (like the one you were suggesting) and _conditional_ iterators (like loop until a certain condition is met).

Reply:
Done! Let's revise this maybe later one once we have the default iterator implementation in.